plugins/ipam/dhcp/lease.go

```diff
@@ -72,29 +72,34 @@ func AcquireLease(clientID, netns, ifName string) (*DHCPLease, error) {
 	log.Printf("%v: acquiring lease", clientID)
 
 	l.wg.Add(1)
-	go ns.WithNetNSPath(netns, true, func(_ *os.File) (e error) {
-		defer l.wg.Done()
-
-		link, err := netlink.LinkByName(ifName)
-		if err != nil {
-			errCh <- fmt.Errorf("error looking up %q", ifName)
-			return
-		}
+	go func() {
+		err := ns.WithNetNSPath(netns, true, func(_ *os.File) (e error) {
```

Comment:
The tricky part here is that in case of error, that closure returns. If it succeeds in getting a lease, it lingers around to keep the lease maintained. `l.maintain()` doesn't return until it gets signaled by `l.Stop()`